props/saber_BC_buttons.h

```diff
@@ -386,36 +1161,254 @@ EFFECT(volmax);     // for maximum volume reached
+  }
+
+  void next() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ < max_volume_) {
+      current_volume_ += max_volume_ * 0.10;
+      if (current_volume_ > max_volume_) current_volume_ = max_volume_;
+      dynamic_mixer.set_volume(current_volume_);
+      mode::getSL<SPEC>()->SayVolumeUp();
+      announced_ = false;
+    } else if (!announced_) {
+      QuickMaxVolume();
+      announced_ = true;
+    }
+  }
+
+  void prev() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ > min_volume_) {
+      current_volume_ -= max_volume_ * 0.10;
+      if (current_volume_ < min_volume_) current_volume_ = min_volume_;
+      dynamic_mixer.set_volume(current_volume_);
+      mode::getSL<SPEC>()->SayVolumeDown();
+      announced_ = false;
+    } else if (!announced_) {
+      QuickMinVolume();
+      announced_ = true;
+    }
+  }
+
+  void QuickMaxVolume() {
+    dynamic_mixer.set_volume(max_volume_);
+    PVLOG_NORMAL << "** Maximum Volume\n";
+    mode::getSL<SPEC>()->SayMaximumVolume();
+  }
+
+  void QuickMinVolume() {
+    dynamic_mixer.set_volume(min_volume_);
+    PVLOG_NORMAL << "** Minimum Volume\n";
+    mode::getSL<SPEC>()->SayMinimumVolume();
+  }
+
+  void update() override {  // Overridden to substitute the tick sound
+    float volume = dynamic_mixer.get_volume();
+    percentage_ = round((volume / max_volume_) * 10) * 10;
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+  }
+
+  void select() override {
+    PVLOG_NORMAL << "** Saved - Exit Volume Menu\n";
+    mode::getSL<SPEC>()->SaySave();
+    SPEC::SteppedMode::select();
+  }
+
+  void exit() override {
+    PVLOG_NORMAL << "** Cancelled - Exit Volume Menu\n";
+    percentage_ = initial_percentage_;
+    dynamic_mixer.set_volume(initial_volume_);
+    mode::getSL<SPEC>()->SayCancel();
+    SPEC::SteppedMode::exit();
+  }
+
+  bool mode_Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+    switch (EVENTID(button, event, 0)) {
+      // Custom button controls for BCVolumeMode
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_HELD_MEDIUM, 0):
+        QuickMaxVolume();
+        return true;
+
+      case EVENTID(BUTTON_POWER, EVENT_SECOND_HELD_MEDIUM, 0):
+      case EVENTID(BUTTON_AUX, EVENT_FIRST_HELD_MEDIUM, 0):
+        QuickMinVolume();
+        return true;
+    }
+    // Use the select and exit controls from SelectCancelMode
+    return SPEC::SelectCancelMode::mode_Event2(button, event, modifiers);
+  }
+};
+
+#ifdef DYNAMIC_BLADE_LENGTH
+
+template<class SPEC>
+struct BCSelectBladeMode : public SPEC::MenuBase {
+  int current_blade() { return this->pos_ + 1; }
+  ShowColorSingleBladeTemplate<Pulsing<White,Black,800>,Pulsing<White,Black,800>> highlighted_blade_;
+  bool single_blade_adjusted_ = false;
+  uint16_t size() override { return NUM_BLADES; }
+
+  void mode_activate(bool onreturn) override {
+    // If only one blade, and returning here via popMode, just exit.
```

Comment:
if there is only one blade, why enter this menu at all?

Reply:
I guess I still don't get pushing and popping so well.
When finishing BCChangeBladeLengthBlade1, it pops back to BCSelectBladeMode, even if we pushed directly to BCChangeBladeLengthBlade1 because NUM_BLADES ==1.
```cpp
#if NUM_BUTTONS > 1
        pushMode<MKSPEC<BCMenuSpec>::BCSelectBladeMenu>();
#else
        pushMode<MKSPEC<BCMenuSpec>::ChangeBladeLengthMode>();
#endif
```
If you mean to exit differently based on NUM_BLADES, I'm going to need a hint.

Reply:
Wait. No it doesn't pop back to BCSelectBladeMode. This whole part can go away.
Done.